dace/sdfg/sdfg.py

```diff
@@ -937,6 +937,12 @@ def free_symbols(self) -> Set[str]:
         # Add free state symbols
         for state in self.nodes():
             free_syms |= state.free_symbols
+            # Add defined nested SDFG symbols
+            for n in state.nodes():
+                if isinstance(n, dace.nodes.NestedSDFG):
+                    for s in n.sdfg.symbols:
+                        if s not in n.symbol_mapping.keys():
+                            defined_syms.add(s)
```

Comment:
That does not make sense to me. The free symbols in the nested SDFG, if not defined in symbol mapping, should not propagate upwards to the higher SDFG

Reply:
Reverted.